arangod/RestHandler/RestReplicationHandler.cpp

```diff
@@ -2126,43 +2131,22 @@ void RestReplicationHandler::handleCommandAddFollower() {
   // get into trouble here we may need to be more lenient
   TRI_ASSERT(checksumSlice.isString() && readLockIdSlice.isString());
 
-  const std::string readLockId = readLockIdSlice.copyString();
+  aql::QueryId readLockId = ExtractReadlockId(readLockIdSlice);
   const std::string checksum = checksumSlice.copyString();
 
-  std::string referenceChecksum;
-  {
-    CONDITION_LOCKER(locker, _condVar);
-    auto it = _holdReadLockJobs.find(readLockId);
-    if (it == _holdReadLockJobs.end()) {
-      // Entry has been removed since, so we cancel the whole thing
-      // right away and generate an error:
-      generateError(rest::ResponseCode::SERVER_ERROR,
-                    TRI_ERROR_TRANSACTION_INTERNAL,
-                    "read transaction was cancelled");
-      return;
-    }
-
-    auto trx = it->second;
-    if (!trx) {
-      generateError(rest::ResponseCode::SERVER_ERROR,
-                    TRI_ERROR_TRANSACTION_INTERNAL,
-                    "Read lock not yet acquired!");
-      return;
-    }
-
-    // referenceChecksum is the stringified number of documents in the
-    // collection
-    uint64_t num = col->numberDocuments(trx.get(), transaction::CountType::Normal);
-    referenceChecksum = std::to_string(num);
+  auto referenceChecksum = computeCollectionChecksum(readLockId, col.get());
```

Comment:
This auto makes the code more difficult to read and understand. I do not insist on removing it, but I would prefer it if the type would be spelled out explicitly here.